src/App/App.js

```diff
@@ -35,10 +38,24 @@ class App extends Component {
     }
   }
 
-  showMovieDetails(id) {
-    this.setState({
+  findUserRatingsForFoundMovie() {
+    const userRating = this.state.userRatings.ratings.find(rating => {
+      return this.state.foundMovieId.id === rating.movie_id
+    })
+    if (userRating === undefined && this.state.isLoggedIn) {
+      this.setState({foundMovieRating: 'You haven\t rated this movie yet'})
+    } else if (!this.state.isLoggedIn) {
+      this.setState({foundMovieRating: 'Log in to rate this movie'})
+    } else {
+      this.setState({foundMovieRating: `You rated this movie: ${userRating.rating}`})
+    }
+  }
```

Comment:
This foundMovieRating is getting passed down to the moviePage component. It might make more sense to move this logic into the moviePage component. The necessary information can be passed into the moviePage component, like the selected movie id and the userRatings in order for the moviePage component to figure out it's own display message.

Reply:
Thanks so much!! This is super helpful & helped clear out the unnecessary `async await` from the `showMovieDetails` method